packages/ckeditor5-core/src/editor/utils/securesourceelement.ts

```diff
@@ -18,16 +18,10 @@ import type { ElementApi } from './elementapimixin';
  * Running multiple editor instances on the same source element causes various issues and it is
  * crucial this helper is called as soon as the source element is known to prevent collisions.
  *
- * @param {module:core/editor/editor~Editor} editor Editor instance.
+ * @param editor Editor instance.
+ * @param sourceElement Element to bind with the editor instance.
  */
-export default function secureSourceElement( editor: Editor & ElementApi ): void {
-	const sourceElement: any = editor.sourceElement;
-
-	// If the editor was initialized without specifying an element, we don't need to secure anything.
-	if ( !sourceElement ) {
-		return;
-	}
-
+export default function secureSourceElement( editor: Editor, sourceElement: HTMLElement & { ckeditorInstance?: Editor } ): void {
```

Comment:
I decided to change this implementation because I don't see a reason why not to pass the `sourceElement` as a parameter. AFAIR, all `secureSourceElement` uses were straight after setting `editor.sourceElement`.

For multi-root, though, there's no `editor.sourceElement` -- there can be multiple. So, the new implementation fits the multi-root editor better.